File: backend/components/api/wire_types/src/resources/org_frequent_note.rs
use crate::*;
use newtypes::TenantFrequentNoteId;
use newtypes::TenantFrequentNoteKind;

/// Frequently used note for an Org
#[derive(Debug, Clone, Serialize, Apiv2Response, macros::JsonResponder)]
#[serde(rename_all = "snake_case")]
pub struct OrgFrequentNote {
    pub id: TenantFrequentNoteId,
    pub kind: TenantFrequentNoteKind,
    pub content: String,
}

File: frontend/packages/types/src/api/org-frequent-note.ts
import type { OrgFrequentNote, OrgFrequentNoteKind } from '../data/org-frequent-note';

export type GetOrgFrequentNotesResponse = Array<OrgFrequentNote>;

export type CreateOrgFrequentNoteRequest = {
  kind: OrgFrequentNoteKind;
  content: string;
};

export type CreateOrgFrequentNoteResponse = OrgFrequentNote;
